test/e2e/metrics_k8s_cluster_receiver_test.go

```diff
@@ -0,0 +1,154 @@
+				g.Expect(resp).To(HaveHTTPStatus(http.StatusOK))
+				bodyContent, err := io.ReadAll(resp.Body)
+				defer resp.Body.Close()
+				g.Expect(err).NotTo(HaveOccurred())
+
+				containerMetricNames := slices.Concat(kubeletstats.ContainerMetricsNames, k8scluster.ContainerMetricsNames)
+				g.Expect(bodyContent).To(WithFlatMetrics(WithNames(
+					OnlyContainElementsOf(containerMetricNames))), "Found container metrics in backend that are not part of k8scluster or kubeletstats")
+
+				g.Expect(bodyContent).To(WithFlatMetrics(
+					ContainElement(HaveField("Name", BeElementOf(k8scluster.ContainerMetricsNames)))), "Found container metrics in backend that are not part of k8scluster")
+
+				g.Expect(bodyContent).To(WithFlatMetrics(
+					ContainElement(HaveField("Name", BeElementOf(kubeletstats.ContainerMetricsNames)))), "Found container metrics in backend that are not part of kubeletstats")
+
+				g.Expect(bodyContent).To(WithFlatMetrics(WithScopeAndVersion(ContainElement(And(
+					HaveField("Name", metric.InstrumentationScopeK8sCluster),
+					HaveField("Version",
+						SatisfyAny(
+							ContainSubstring("main"),
+							ContainSubstring("1."),
+							ContainSubstring("PR-"),
+						)))))), "Only scope '%v' must be sent to the runtime backend", metric.InstrumentationScopeK8sCluster)
+
+			}, periodic.TelemetryEventuallyTimeout, periodic.TelemetryInterval).Should(Succeed())
+		})
+
+		It("Should deliver runtime pod metrics to pod-metrics backend", func() {
+			Eventually(func(g Gomega) {
+				resp, err := proxyClient.Get(backendK8sClusterPodMetricsEnabledURL)
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(resp).To(HaveHTTPStatus(http.StatusOK))
+				bodyContent, err := io.ReadAll(resp.Body)
+				defer resp.Body.Close()
+				g.Expect(err).NotTo(HaveOccurred())
+
+				podMetricNames := slices.Concat(kubeletstats.PodMetricsNames, k8scluster.PodMetricsNames)
+
+				g.Expect(bodyContent).To(WithFlatMetrics(WithNames(
+					OnlyContainElementsOf(podMetricNames))), "Found pod metrics in backend that are not part of k8scluster or kubeletstats")
+
+				g.Expect(bodyContent).To(WithFlatMetrics(
+					ContainElement(HaveField("Name", BeElementOf(k8scluster.PodMetricsNames)))), "Found pod metrics in backend that are not part of k8scluster")
+
+				g.Expect(bodyContent).To(WithFlatMetrics(
+					ContainElement(HaveField("Name", BeElementOf(kubeletstats.PodMetricsNames)))), "Found pod metrics in backend that are not part of kubeletstats")
+
+				g.Expect(bodyContent).To(WithFlatMetrics(WithScopeAndVersion(ContainElement(And(
+					HaveField("Name", metric.InstrumentationScopeK8sCluster),
+					HaveField("Version",
+						SatisfyAny(
+							ContainSubstring("main"),
+							ContainSubstring("1."),
+							ContainSubstring("PR-"),
+						)))))), "Only scope '%v' must be sent to the runtime backend", metric.InstrumentationScopeK8sCluster)
+
+			}, periodic.TelemetryEventuallyTimeout, periodic.TelemetryInterval).Should(Succeed())
+		})
```

Comment:
Similar to the metrics_runtime_input_resources_test.go](https://github.com/kyma-project/telemetry-manager/blob/3c3d15ff2eec2da4072ebaa5e6be51723981f73a/test/e2e/metrics_runtime_input_resources_test.go), can we add 2 tests `Should not deliver runtime pod metrics to container-metrics backend` and `Should not deliver runtime container metrics to pod-metrics backend`  for the k8scluster receiver metrics? Because now we are not testing the drop filter in e2e setup

Reply:
removed the test case